### PR TITLE
Iris dataset documentation

### LINKED ISSUE #447: Iris dataset UCI Machine Learning Repo Note
Add a note that two data points in UCI's Machine Learning repo are different from the Iris dataset in R (and the one in the original Fischer paper)

### PR DESCRIPTION
**Pull request by janismdhanbad**
### Description
Adding note about the difference in Iris dataset in UCI repo and R. 

### Related issues or pull requests

Fixes #447 

### Pull Request Checklist

- [x] Added a note about the modification or contribution to the `./docs/sources/CHANGELOG.md` file (if applicable)
- [x] Added appropriate unit test functions in the `./mlxtend/*/tests` directories (if applicable)
- [x] Modify documentation in the corresponding Jupyter Notebook under `mlxtend/docs/sources/` (if applicable)
- [x] Ran `nosetests ./mlxtend -sv` and make sure that all unit tests pass (for small modifications, it might be sufficient to only run the specific test file, e.g., `nosetests ./mlxtend/classifier/tests/test_stacking_cv_classifier.py -sv`)
- [x] Checked for style issues by running `flake8 ./mlxtend`

### CODE REVIEW

File: mlxtend/data/iris.py [hunk 1]
```diff
@@ -36,6 +36,15 @@ def iris_data():
...
         y is a 1-dimensional array of the class labels {0, 1, 2}
 
+
+    Note
+    --------
+    Iris dataset present in UCI's machine learning repository is different
+    from Iris data availble in R (and the one in the original Fischer paper).
+    Precisely, there are two data points(row number 34 and 37) in UCI's Machine
+    Learning repo are different from the Iris dataset in R (and the one in the
+    original Fischer paper).
```
Comment (rasbt) [inline mlxtend/data/iris.py]: "Fischer should be Fisher" :)

I suggest rewording

```
Iris dataset present in UCI's machine learning repository is different
  from Iris data availble in R (and the one in the original Fischer paper).
  Precisely, there are two data points(row number 34 and 37) in UCI's Machine
  Learning repo are different from the Iris dataset in R (and the one in the
  original Fischer paper).
```

to 

```
The Iris dataset (originally collected by Edgar Anderson) and available in 
UCI's machine learning repository is different
from the Iris dataset available in R 
(and the one in the original paper by R.A. Fisher [1]).
Precisely, there are two data points(row number 34 and 37) in UCI's Machine
Learning repository are different from the Iris dataset in R (and the one in the
original Fisher paper).

[1] . A. Fisher (1936). "The use of multiple measurements in taxonomic problems". 
Annals of Eugenics. 7 (2): 179–188

```

File: docs/sources/CHANGELOG.md [hunk 2]
```diff
@@ -57,6 +57,7 @@ The CHANGELOG for the current development version is available at
 - Raises a more informative error if a `SparseDataFrame` is passed to `apriori` and the dataframe has integer column names that don't start with `0` due to current limitations of the `SparseDataFrame` implementation in pandas. ([#503](https://github.com/rasbt/mlxtend/pull/503))
 - SequentialFeatureSelector now supports DataFrame as input for all operating modes (forward/backward/floating). [#506](https://github.com/rasbt/mlxtend/pull/506)
 - `mlxtend.evaluate.feature_importance_permutation` now correctly accepts scoring functions with proper function signature as `metric` argument. [#528](https://github.com/rasbt/mlxtend/pull/528)
+- Fixed documentation of `iris_data()` under `iris.py` by adding a note about differences in the iris data in R and UCI machine learning repo.
```
Comment (rasbt) [inline docs/sources/CHANGELOG.md]: Sorry, this is maybe a bit of an awkward timing, but I just made a new release version this morning. Could you add this line to the 0.17.0dev0 section at the top of the document instead?

File: mlxtend/data/iris.py (hunk 1 above)
Comment (janismdhanbad) [reply mlxtend/data/iris.py]: ### Hi @rasbt, I have made the required changes.

File: docs/sources/CHANGELOG.md (hunk 2 above)
Comment (janismdhanbad) [reply docs/sources/CHANGELOG.md]: Hi @rasbt, I have made the required changes.

**Comment by rasbt** [comment]
Thanks, looks good to me now! Are you also planning to add the suggestion regarding

- iris_data(version='uci') (default)
- and iris_data(version='R')

in this PR, or shall I make a separate issue for that?

**Comment by rasbt** [comment]
Btw for this part

> Modify documentation in the corresponding Jupyter Notebook under mlxtend/docs/sources/ (if applicable)

what would be required is

- cd into the mlxtend/docs folder
- run `python make_api.py`
- open the jupyter notebook in docs/sources/user_guide
- click "run all cells" and save the notebook

**Comment by janismdhanbad** [comment]
Thanks for the steps to check 

> Modify documentation in the corresponding Jupyter Notebook under mlxtend/docs/sources/ (if applicable)

I have made those checks and successfully committed the code. 

Regarding the following changes:
* iris_data(version='uci') (default)
* and iris_data(version='R')

I think it would be better to make a separate issue. If you could please tag me in that issue, I will take that up then.

File: docs/sources/user_guide/data/iris_data.ipynb [hunk 3]
```diff
@@ -71,13 +71,14 @@
...
    "outputs": [
     {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "Dimensions: 150 x 4\n",
-      "\n",
-      "Header: ['sepal length', 'sepal width', 'petal length', 'petal width']\n",
-      "1st row [ 5.1  3.5  1.4  0.2]\n"
+     "ename": "ModuleNotFoundError",
```
Comment (rasbt) [inline docs/sources/user_guide/data/iris_data.ipynb]: Looks like there's an issue with your mlxtend installation. I recommend running

```
python setup.py develop`
```

in the main folder of your mlxtend branch again, then try to run `python make_api.py` again, followed by running the notebook again.

**Comment by janismdhanbad** [comment]
Hi, I have made the changes but nosetest is giving me 

`

> Split arrays or matrices into random train and test subsets ... ERROR

`

this time. when I am running it on the dev branch.

**Comment by rasbt** [comment]
> Split arrays or matrices into random train and test subsets ... ERROR

Hm, that's weird. Let's see how the CI tests turn out

**Comment by rasbt** [comment]
Looks fine to me! Will merge. Thanks for the contribution!
